propsigner/streamproto.go

```diff
@@ -0,0 +1,55 @@
+package propsigner
+
+import (
+	"errors"
+	"io"
+
+	"github.com/multiformats/go-varint"
+	"google.golang.org/protobuf/proto"
+)
+
+type byteReader struct {
+	r io.Reader
+}
+
+func (d *byteReader) ReadByte() (byte, error) {
+	var buf [1]byte
+	_, err := d.r.Read(buf[:])
+	return buf[0], err
+}
+
+func readMsg(r io.Reader, maxSize int, msg proto.Message) error {
+	mlen, err := varint.ReadUvarint(&byteReader{r: r})
+	if err != nil {
+		return err
+	}
+
+	buf := make([]byte, maxSize)
+	if uint64(len(buf)) < mlen {
```

Comment:
Nit: this check can happen before allocating memory.

Reply:
Will move :+1: